### PR TITLE
Register mbeans in custom properties 

### LINKED ISSUE #19085: Is there any way to monitor the connection pool
## Question

Is there any way to monitor the connection pool like druid monitor？

Recently, I have encountered several times in the online environment that the sharding proxy database connection is unavailable, which will occur one day to one month after the server is started.

At this time, I try to use Navicat to test the direct connection to MySQL. It can connect, but it can't connect to sharding proxy.

The log is as follows:
```
### Error querying database.  Cause: org.springframework.jdbc.CannotGetJdbcConne
ctionException: Failed to obtain JDBC Connection; nested exception is java.sql.S
QLTransientConnectionException: HikariPool-1 - Connection is not available, requ
est timed out after 30001ms.
### The error may exist in com/xxx/dao/XXDao.java (b
est guess)
### The error may involve com.xxx.dao.XXDao.selectXXByID
### The error occurred while executing a query
### Cause: org.springframework.jdbc.CannotGetJdbcConnectionException: Failed to
obtain JDBC Connection; nested exception is java.sql.SQLTransientConnectionExcep
tion: HikariPool-1 - Connection is not available, request timed out after 30001m
s.
```
I hope there is a way to monitor the status of sharding proxy connection pool. Can you give me some suggestions.

Thanks.

### PR DESCRIPTION
**Pull request by sunkai-cai**
Fixes #19085
related issue #15695
another PR #19904

Changes proposed in this pull request:
- support Hikari to use JMX  

```
dataSources:
  ds_0:
    dataSourceClassName: com.zaxxer.hikari.HikariDataSource
    url: jdbc:mysql://local-mysql:3306/db_0?serverTimezone=UTC&useSSL=false&characterEncoding=utf-8
    username: sphere
    password: sphere
    customPoolProps:
      registerMbeans: true
```

### CODE REVIEW

File: shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/datasource/pool/metadata/DataSourcePoolMetaData.java [hunk 1]
```diff
@@ -50,6 +50,13 @@ public interface DataSourcePoolMetaData extends TypedSPI {
      */
     Map<String, String> getPropertySynonyms();
     
+    /**
+     * Get jdbc url synonym field names.
+     *
+     * @return jdbc url synonym field names
+     */
+    Collection<String> getJdbcUrlSynonymFieldNames();
```
Comment (TeslaCN) [inline shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/datasource/pool/metadata/DataSourcePoolMetaData.java:58]: Why name it `JdbcUrlSynonymFieldNames`? It seems that `registerMbeans` has nothing to do with JDBC URL.

Comment (sunkai-cai) [reply shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/datasource/pool/metadata/DataSourcePoolMetaData.java:58]: `JdbcUrlPropertiesFieldName`  is from `DataSourcePoolFieldMetaData.getJdbcUrlPropertiesFieldName`  for interface

Comment (sunkai-cai) [reply shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/datasource/pool/metadata/DataSourcePoolMetaData.java:58]:
```
public final class HikariDataSourcePoolFieldMetaData implements DataSourcePoolFieldMetaData {
    ...
    ...
    @Override
    public String getJdbcUrlPropertiesFieldName() {
        return "dataSourceProperties";
    }
}
```

Comment (sunkai-cai) [reply shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/datasource/pool/metadata/DataSourcePoolMetaData.java:58]: replace  `JdbcUrlProperties`  with `customPoolProperties`  is better I think

**Review by TeslaCN — CHANGES_REQUESTED** [review]
Before further reviewing, may I ask what is this PR for?
I tried the following configuration and found it worked.

```yaml
schemaName: bmsql_sharding
dataSources:
  ds_0:
    url: jdbc:postgresql://127.0.0.1:5432/bmsql_0
    username: postgres
    password: postgres
    connectionTimeoutMilliseconds: 3000
    idleTimeoutMilliseconds: 60000
    maxLifetimeMilliseconds: 1800000
    maxPoolSize: 32
    minPoolSize: 0
    customPoolProps:
      registerMbeans: true
  ds_1:
    url: jdbc:postgresql://127.0.0.1:5432/bmsql_1
    username: postgres
    password: postgres
    connectionTimeoutMilliseconds: 3000
    idleTimeoutMilliseconds: 60000
    maxLifetimeMilliseconds: 1800000
    maxPoolSize: 32
    minPoolSize: 0
    customPoolProps:
      registerMbeans: true
  ds_2:
    url: jdbc:postgresql://127.0.0.1:5432/bmsql_2
    username: postgres
    password: postgres
    connectionTimeoutMilliseconds: 3000
    idleTimeoutMilliseconds: 60000
    maxLifetimeMilliseconds: 1800000
    maxPoolSize: 32
    minPoolSize: 0
    customPoolProps:
      registerMbeans: true
  ds_3:
    url: jdbc:postgresql://127.0.0.1:5432/bmsql_3
    username: postgres
    password: postgres
    connectionTimeoutMilliseconds: 3000
    idleTimeoutMilliseconds: 60000
    maxLifetimeMilliseconds: 1800000
    maxPoolSize: 32
    minPoolSize: 0
    customPoolProps:
      registerMbeans: true
```

image

**Comment by sunkai-cai** [comment]
@TeslaCN  yes, you are right.  thanks for your tips.

**Pull request closed without merging**